### PR TITLE
Add RISC-V 64-bit container builds

### PR DESCRIPTION
**Pull request by MichaIng**
## Thank you for your contribution to the Pi-hole Community!

Please read the comments below to help us consider your Pull Request.

We are all volunteers and completing the process outlined will help us review your commits quicker.

**Please make sure you**

 1. Base your code and PRs against the repositories developmental branch.
 2. [Sign Off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits as we enforce the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
 3. [Sign](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all your commits as they must have verified signatures
 4. File a pull request for any change that requires changes to [our documentation](https://docs.pi-hole.net/) at our [documentation repo](https://github.com/pi-hole/docs) 

---

**What does this PR aim to accomplish?:**

Adding RISC-V 64-bit container builds.

**How does this PR accomplish the above?:**

- Added `riscv64` Dockerfile
- Added `riscv64` target to FTL builds and base image workflows.
- _Removed doubled `git` package from `aarch64` Dockerfile_
- [x] Decide whether it makes sense to use a Debian Buster container base image, as RISC-V is currently only support on Debian ports Sid/unstable.
    **EDIT**: Using Debian Sid base container now, since `libc6-dev:riscv64` depends on recent versions of glibc packages anyway, which forcefully upgrades a Buster system to Sid in relevant parts anyway. On case builds for older glibc are needed, flags would need to be passed.
- [ ] Tests/builds fail since GitHub secrets are no available for actions from forks.
    **EDIT**: Tested the steps manually on a Sid VM with https://github.com/pi-hole/FTL/pull/1536 and the build goes through and produces an executable FTL which can be started and shows CLI via `qemu-riscv64-static`. Adjusted binary tests in https://github.com/pi-hole/FTL/pull/1536 to match the results. Test on physical RISC-V system outstanding.

Companion PRs:
- https://github.com/pi-hole/FTL/pull/1536
- https://github.com/pi-hole/pi-hole/pull/5196

**Link documentation PRs if any are needed to support this PR:**

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [x] I have read the above and my PR is ready for review. *Check this box to confirm*

### CODE REVIEW

**Comment by DL6ER** [comment]
Have you tested some of the stand-alone features of FTL on your hardware to see if it actually runs when built by this container? I'm thinking about some easy things:

- `pihole-FTL sqlite3`
- `pihole-FTL lua`
- `sudo pihole-FTL dhcp-discover`
- `pihole-FTL dnsmasq-test`

---

Others that that: I've tried building your docker container locally but it failed with:
```
Hit:1 http://deb.debian.org/debian sid InRelease
Ign:2 https://deb.debian.org/debian-ports sid InRelease
Ign:2 https://deb.debian.org/debian-ports sid InRelease
Ign:2 https://deb.debian.org/debian-ports sid InRelease
Err:2 https://deb.debian.org/debian-ports sid InRelease
  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: 146.75.122.132 443]
Reading package lists...
W: https://deb.debian.org/debian-ports/dists/sid/InRelease: No system certificates available. Try installing ca-certificates.
W: https://deb.debian.org/debian-ports/dists/sid/InRelease: No system certificates available. Try installing ca-certificates.
W: https://deb.debian.org/debian-ports/dists/sid/InRelease: No system certificates available. Try installing ca-certificates.
W: https://deb.debian.org/debian-ports/dists/sid/InRelease: No system certificates available. Try installing ca-certificates.
W: Failed to fetch https://deb.debian.org/debian-ports/dists/sid/InRelease  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: 146.75.122.132 443]
W: Some index files failed to download. They have been ignored, or old ones used instead.
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package libcap-dev:riscv64
E: Unable to locate package libgmp-dev:riscv64
E: Unable to locate package libc6-dev:riscv64
```

The hint works, when moving `ca-certificates` up to where `debian-ports-archive-keyring` is installed, it works fine.

---

The next error it is getting stuck in is:
```
dpkg: error processing archive /tmp/apt-dpkg-install-f4hSKS/104-gcc-12-base_12.2.0-14_riscv64.deb (--unpack):
 package architecture (riscv64) does not match system (amd64)

[...]

dpkg: error processing archive /tmp/apt-dpkg-install-f4hSKS/121-libgcc-s1_12.2.0-14_riscv64.deb (--unpack):
 package architecture (riscv64) does not match system (amd64)
dpkg: error processing archive /tmp/apt-dpkg-install-f4hSKS/122-libc6_2.36-8_riscv64.deb (--unpack):
 package architecture (riscv64) does not match system (amd64)
dpkg: error processing archive /tmp/apt-dpkg-install-f4hSKS/123-linux-libc-dev_6.1.12-1_riscv64.deb (--unpack):
 package architecture (riscv64) does not match system (amd64)
dpkg: error processing archive /tmp/apt-dpkg-install-f4hSKS/124-libcrypt1_1%3a4.4.33-2_riscv64.deb (--unpack):
 package architecture (riscv64) does not match system (amd64)
dpkg: error processing archive /tmp/apt-dpkg-install-f4hSKS/125-libcrypt-dev_1%3a4.4.33-2_riscv64.deb (--unpack):
 package architecture (riscv64) does not match system (amd64)
dpkg: error processing archive /tmp/apt-dpkg-install-f4hSKS/126-libcom-err2_1.47.0-2_riscv64.deb (--unpack):
 package architecture (riscv64) does not match system (amd64)
dpkg: error processing archive /tmp/apt-dpkg-install-f4hSKS/127-libkrb5support0_1.20.1-1_riscv64.deb (--unpack):
 package architecture (riscv64) does not match system (amd64)
```
(and many similar others)

I'm running these test on an `x86_64` machine with `Ubuntu 22.04.2 LTS` and `docker` version `20.10.14, build a224086`. The base image (`debian:sid`) pulled in my experiments in `bad0dd70f005` (4 days ago).

File: ftl-build/riscv64/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,87 @@
+FROM debian:sid AS builder
+
+ARG idnversion=1.41
+ARG readlineversion=8.1
+ARG termcapversion=1.3.1
+ARG nettleversion=3.8.1
+
+RUN apt-get update \
+    && apt-get install debian-ports-archive-keyring \
```
Comment (DL6ER) [inline ftl-build/riscv64/Dockerfile:9]:
```suggestion
    && apt-get install --no-install-recommends -y ca-certificates debian-ports-archive-keyring \
```

File: ftl-build/riscv64/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    && echo 'deb [arch=riscv64] https://deb.debian.org/debian-ports sid main' > /etc/apt/sources.list.d/riscv64.list \
+    && dpkg --add-architecture riscv64 \
+    && apt-get update \
+    && apt-get install --no-install-recommends -y \
+        ca-certificates \
```
Comment (DL6ER) [inline ftl-build/riscv64/Dockerfile:14]:
```suggestion
```

**Comment by MichaIng** [comment]
The part solved with initially missing `ca-certificates`.

> Have you tested some of the stand-alone features of FTL on your hardware to see if it actually runs when built by this container?

So far only some quick tests with `qemu-riscv64-static` on the machine where I tested the built (Debian Bookworm VM with Sid repo added). I uploaded the binary already for testing on the (remote) StarFive VisionFive 2.

**Comment by MichaIng** [comment]
Tested, works, here a debug token to have a closer look: https://tricorder.pi-hole.net/kcm5p3sW/

**Review by DL6ER — APPROVED** [review]
The debug log looks fine. Excellent job, thanks!

**Comment by dschaper** [comment]
@MichaIng Do you have IPv6 disabled (via `sysctl.conf` or `grub`) on that test setup?

**Comment by MichaIng** [comment]
The kernel does not support it at all:
```sh
root@DietPi:~# modinfo ipv6
modinfo: ERROR: Module ipv6 not found.
```
This is the kernel provided by StarFive, which lacks quick some features. I'll try doing an own build next week.

**Comment by MichaIng** [comment]
Generally `buildx` supports the `linux/riscv64` target: https://github.com/docker/buildx/pull/652

Would make sense if it required a Sid base container here as well.

**Pull request merged**